daemon/logger/etwlogs/etwlogs_windows.go

```diff
@@ -146,7 +146,13 @@ func callEventRegister() error {
 }
 
 func callEventWriteString(message string) error {
-	ret, _, _ := procEventWriteString.Call(uintptr(providerHandle), 0, 0, uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(message))))
+	utf16message, err := syscall.UTF16FromString(message)
```

Comment:
I expand `StringToUTF16Ptr` to `UTF16FromString`

the `StringToUTF16Ptr` code can be found in <https://golang.org/src/syscall/syscall_windows.go>

```
func StringToUTF16Ptr(s string) *uint16 { return &StringToUTF16(s)[0] }

func StringToUTF16(s string) []uint16 {
	a, err := UTF16FromString(s)
	if err != nil {
		panic("syscall: string with NUL passed to StringToUTF16")
	}
	return a
}
```